core/templates/pages/exploration-editor-page/editor-tab/services/responses.service.ts

```diff
@@ -122,35 +122,45 @@ angular.module('oppia').factory('ResponsesService', [
 
     var _updateAnswerGroup = function(index, updates, callback) {
       var answerGroup = _answerGroups[index];
-      if (updates.hasOwnProperty('rules')) {
-        answerGroup.rules = updates.rules;
-      }
-      if (updates.hasOwnProperty('taggedSkillMisconceptionId')) {
-        answerGroup.taggedSkillMisconceptionId =
-          updates.taggedSkillMisconceptionId;
-      }
-      if (updates.hasOwnProperty('feedback')) {
-        answerGroup.outcome.feedback = updates.feedback;
-      }
-      if (updates.hasOwnProperty('dest')) {
-        answerGroup.outcome.dest = updates.dest;
-      }
-      if (updates.hasOwnProperty('refresherExplorationId')) {
-        answerGroup.outcome.refresherExplorationId = (
-          updates.refresherExplorationId);
-      }
-      if (updates.hasOwnProperty('missingPrerequisiteSkillId')) {
-        answerGroup.outcome.missingPrerequisiteSkillId = (
-          updates.missingPrerequisiteSkillId);
-      }
-      if (updates.hasOwnProperty('labelledAsCorrect')) {
-        answerGroup.outcome.labelledAsCorrect = updates.labelledAsCorrect;
-      }
-      if (updates.hasOwnProperty('trainingData')) {
-        answerGroup.trainingData = updates.trainingData;
+
+      if (answerGroup) {
+        if (updates.hasOwnProperty('rules')) {
+          answerGroup.rules = updates.rules;
+        }
+        if (updates.hasOwnProperty('taggedSkillMisconceptionId')) {
+          answerGroup.taggedSkillMisconceptionId =
+            updates.taggedSkillMisconceptionId;
+        }
+        if (updates.hasOwnProperty('feedback')) {
+          answerGroup.outcome.feedback = updates.feedback;
+        }
+        if (updates.hasOwnProperty('dest')) {
+          answerGroup.outcome.dest = updates.dest;
+        }
+        if (updates.hasOwnProperty('refresherExplorationId')) {
+          answerGroup.outcome.refresherExplorationId = (
+            updates.refresherExplorationId);
+        }
+        if (updates.hasOwnProperty('missingPrerequisiteSkillId')) {
+          answerGroup.outcome.missingPrerequisiteSkillId = (
+            updates.missingPrerequisiteSkillId);
+        }
+        if (updates.hasOwnProperty('labelledAsCorrect')) {
+          answerGroup.outcome.labelledAsCorrect = updates.labelledAsCorrect;
+        }
+        if (updates.hasOwnProperty('trainingData')) {
+          answerGroup.trainingData = updates.trainingData;
+        }
+        _saveAnswerGroups(_answerGroups);
+        callback(_answerGroupsMemento);
+      } else {
+        if (_activeAnswerGroupIndex !== -1) {
+          _activeAnswerGroupIndex = -1;
```

Comment:
Not sure if this is an intended change. I like that you have handled the corner case.. @aks681 could you clarify?

Reply:
Yeah, this change seems fine, shouldn't affect the update process in a normal flow, though maybe remove the `if (_activeAnswerGroupIndex !== -1)` ? Since, that's redundant, right? As by the end of this `_activeAnswerGroupIndex ` is -1 anyway. You can just do `_activeAnswerGroupIndex  = -1` here.

Reply:
Done.